src/lib/mirage_block_ocaml.ml

```diff
@@ -360,16 +371,23 @@ let process_one t =
             return Response.Delete
           | Request.Flush h ->
             let t = Handle.find_or_quit h in
-            Block.flush t.Handle.base
-            >>= fun () ->
-            return Response.Flush
+            let open Lwt.Infix in
+            (* Block.write_error <> Qcow.write_error *)
+            begin Block.flush t.Handle.base
+              >>= function
+              | Error `Disconnected -> Lwt.return (Error `Disconnected)
+              | Error `Is_read_only -> Lwt.return (Error `Is_read_only)
```

Comment:
Sorry I deleted my comment because I thought I'd typo'd something. I've double-checked and it still doesn't work for me, which is probably a sign of a problem in `ocaml-qcow`?
```
File "src/lib/mirage_block_ocaml.ml", line 376, characters 12-432:
Error: This expression has type
         (Protocol.Response.ok, Block.write_error) result Lwt.t
       but an expression was expected of type
         (Protocol.Response.ok, Qcow.write_error) result Lwt.t
       Type
         Block.write_error =
           [> `Disconnected | `Is_read_only | `Unimplemented ]
       is not compatible with type
         Qcow.write_error =
           [> `Disconnected | `Is_read_only | `Unimplemented ] 
```
WDYT?